sklearn/model_selection/tests/test_search.py

```diff
@@ -1846,3 +1839,61 @@ def test_search_cv__pairwise_property_equivalence_of_precomputed():
 
     attr_message = "GridSearchCV not identical with precomputed metric"
     assert (preds_original == preds_precomputed).all(), attr_message
+
+
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'a': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'a': np.random.randint(1, 3, size=2)})]
+)
+def test_scalar_fit_param(SearchCV, param_search):
+    # check general support for scalar in fit_params
+    # non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/15805
+    class TestEstimator(BaseEstimator, ClassifierMixin):
+        def __init__(self, a=None):
+            self.a = a
+
+        def fit(self, X, y, r=None):
+            assert r == 42
+
+        def predict(self, X):
+            return np.zeros(shape=(len(X)))
+
+    model = SearchCV(TestEstimator(), param_search)
+    X, y = make_classification(random_state=42)
+    model.fit(X, y, r=42)
+
+
+def _custom_lgbm_metric(y_test, y_pred):
+    # y_pred are probablities which need to be thresholded
+    y_pred = (y_pred > 0.5).astype(int)
+    acc = accuracy_score(y_test, y_pred)
+    # required output of format: (eval_name, eval_result, is_higher_better)
+    return ('accuracy', acc, True)
+
+
+@pytest.mark.parametrize("metric", ['auc', _custom_lgbm_metric])
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'learning_rate': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'learning_rate': uniform(0.01, 0.1)})]
+)
+def test_scalar_fit_param_lgbm(metric, SearchCV, param_search):
+    # check support for scalar in fit_params in LightGBM
+    # non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/15805
+    lgbm = pytest.importorskip("lightgbm")
+    X_train, X_valid, y_train, y_valid = train_test_split(
+        *make_classification(random_state=42), random_state=42
+    )
+    model = SearchCV(
+        lgbm.LGBMClassifier(n_estimators=5), param_search
+    )
+    fit_params = {
+        'eval_set': [(X_valid, y_valid)],
+        'eval_metric': metric,
+        'early_stopping_rounds': 5,
+        'verbose': False
```

Comment:
Maybe we should add a comment to state that this is an abuse of the scikit-learn Estimator API: officially fit parameters should be data dependent (e.g. `sample_weights`) which is not the case for those. But this abuse is common in popular third party libraries we should tolerate this behavior for now and be careful not to break support for those without following proper deprecation cycle.